### PR TITLE
[JUJU-796] Remove some alias commands

### PR DESCRIPTION
**Pull request by jack-w-shaw**
Here is a full list of the command changes from `juju`:
- Drop `upgrade-charm` in favour of `refresh`
- Drop `remove-consumed-application` in favour of `remove-saas`
- Drop `hook-tool` and `hook-tools` in favour of `help-tool`
- Drop `sync-tools` in favour of `sync-agent-binary`
- Drop `upgrade-juju` in favour of `upgrade-model`
- Drop `attach` in favour of `attach-resource`
- Drop `show-status` in favour of `status`

And from `juju-metadata`
- Drop `generate-tools`/`generate-agents` and change to `generate-agent-binaries`
- Drop `validate-tools`/`validate-agents` and change to `validate-agent-binaries`
- Add `images` to `list-images` for consistency

Also, consistently make list-* variants aliases where applicable, and drop the `list-` alias from examples

## Checklist

 - ~[ ] Requires a [pylibjuju](https://github.com/juju/python-libjuju) change~
 - [x] Added [integration tests](https://github.com/juju/juju/tree/develop/tests) for the PR
 - ~[ ] Added or updated [doc.go](https://discourse.jujucharms.com/t/readme-in-packages/451) related to packages changed~
 - [ ] Comments answer the question of why design decisions were made

## QA steps

```sh
make static-analysis
go test github.com/juju/juju/cmd/juju/...
go test github.com/juju/juju/cmd/plugins/juju-metadata
```

```sh
$ juju show-status
ERROR juju: "show-status" is not a juju command. See "juju --help".
```

## Documentation changes

Probably does require a documentation change or two

## Bug reference

N/A

### CODE REVIEW

File: cmd/juju/firewall/listrules.go [hunk 1]
```diff
@@ -56,7 +56,6 @@ func (c *listFirewallRulesCommand) Info() *cmd.Info {
 		Name:    "list-firewall-rules",
 		Purpose: listRulesHelpSummary,
 		Doc:     listRulesHelpDetails,
-		Aliases: []string{"firewall-rules"},
```
Comment (jack-w-shaw) [inline cmd/juju/firewall/listrules.go:59]: Do we want to keep `firewall-rules` and remove `list-firewall-rules`? That way we'll be consistent with other commands

Comment (jack-w-shaw) [reply cmd/juju/firewall/listrules.go:59]: I see no reason why we should have `list-firewall-rules` as an exception so I've decided to go ahead with this

**Comment by jameinel** [comment]
wallets and plans should just be removed rather than worrying about list-* variants.
I don't remember any discussion about removing list-* variants. We can certainly discuss it.

I don't like `hook-tool` because it is the 'help about hook tools' and it is neither plural nor help. Though the singular also exists for the `juju hook-tool unit-get` (`juju hook-tools unit-get` is awkward, but so is `juju hook-tool` listing all hook tools). I think we should have more discussion here. (I'm personally fine with getting rid of `juju help-tool` )

Hm. maybe I should go one by one, with a caveat that I don't believe we concretely decided to get rid of all `list-*` variants, but those would be taken as a group (either we should get rid of all of them, or leave them all).

 * Drop upgrade-charm in favour of refresh (Is this sensible?)
seems reasonable

 * Drop remove-consumed-application in favour of remove-saas
fine

 * Drop resolve in favour of resolved (Is this sensible?)
fine

 * Drop set-default-credential in favour of default-credential
This one I'm not sure about. `juju default-credential aws myawscred` is concrete, but not quite as clear as `juju set-default-credential aws myawscred`. More discussion I think.

 * Drop set-default-region in favour of default-region
Similar to `set-default-credential`

 * Drop show-credentials in favour of show-credential (is this sensible?)
Personally I think the singular and plural forms are helpful to users here.

 * Drop update-credentials in favour of update-credential (is this sensible?)
same

 * Drop help-tool and hook-tools in favour of hook-tool (is this sensible? A few of references to help-tool in codebase)
I think this needs a bit of design discussion.

 * Drop sync-tools in favour of sync-agent-binaries (is this sensible?)
Yes.

 * Drop upgrade-juju in favour of upgrade-model
Yes

 * Drop model-default in favour of model-defaults (is this sensible?)
It feels awkward when setting one default `juju model-defaults aws foo=bar`, I think the singular is still useful here.

 * Drop list-commits in favour of commits (wtf??)
Ask Joe, but this should be a hidden command anyway (I thought it was only enaled with a feature flag)

 * Drop attach in favour of attach-resource (is this sensible?)
We also have `attach-storage` so we shouldn't reserve the single word, so I think it is good to drop.

 * Drop list-plans in favour of plans
 * Drop list-wallets in favour of wallets
wallets and plans should be dropped completely.

 * Drop debug-hook in favour of debug-hooks (is this sensible?)
I think it helps users if they can `juju debug-hook mongodb install`.

 * Drop show-status in favour of status
Seems ok. I know we had designs around show being more common (show-unit, etc), but I don't know anyone that uses show-status.

And from juju-metadata

    Drop generate-tools in favour of generate-agents (is this sensible?)
    Drop validate-tools in favour of validate-agents (is this sensible?)
Those really should be agent-binaries vs `agents`, though it does make them a bit unweildy (though they are very infrequently used).

**Comment by hloeung** [comment]
Can't keep up with all the command changes. Any chance we can ensure backwards compatibility? We have tools written around as well as documentation and playbooks.

**Comment by sajoupa** [comment]
>     * Drop resolve in favour of resolved (Is this sensible?)
>       fine
I remember a lot of users being confused by "resolved" back when it was the only option. Going back would be quite unpleasant.

**Comment by jnsgruk** [comment]
> Can't keep up with all the command changes. Any chance we can ensure backwards compatibility? We have tools written around as well as documentation and playbooks.

Unfortunately we've been carrying a lot of these for a really long time, and having multiple ways to achieve the same thing breaks any kind of consistency in the command line and makes it difficult for beginners (or not!) to figure out the "right patterns". Note that these changes will land in Juju 3, a major release - which is about the only time we get to make such changes.

> > ```
> > * Drop resolve in favour of resolved (Is this sensible?)
> >   fine
> > ```
> 
> I remember a lot of users being confused by "resolved" back when it was the only option. Going back would be quite unpleasant.

I actually agree here. Most of the commands we have contain a verb/noun combo. `resolved` vs `resolve` seems like a strange trade off - I personally feel like `juju resolve foo` makes more sense.

**Review by anvial — APPROVED** [review]
Need also to check:

https://github.com/juju/juju/blob/abad4560021e133b06205b0867369f950ce3a77d/doc/simplestreams-metadata.txt#L245

Otherwise, QA passed
LG2M

**Pull request merged**